vowpalwabbit/v_array.h

```diff
@@ -151,41 +172,69 @@ struct v_array
   inline size_t size() const { return _end - _begin; }
   inline size_t capacity() const { return end_array - _begin; }
 
-  void resize(size_t length)
+  // maintain the original (deprecated) interface for compatibility. To be removed in VW 10
+  //   VW_DEPRECATED(
+  //       "v_array::resize() is deprecated. Use reserve() instead.
+  // For standard resize behavior, use actual_resize(). The function names will be re-aligned in VW 10")
+  void resize(size_t length) { reserve_nocheck(length); }
```

Comment:
We'll need to destroy objects if the v_array is shrinking. Alternatively we could just call `reserve()` here? Not sure what the expected behavior is

Reply:
I'm not sure the best way to handle this because the old resize did not destroy. We will also not need to destroy once constrained to trivially copyable only. The new actual_resize destroys too.

I think we should keep as is because it is bug for bug compatible with the old versioj